src/libcore/ptr.rs

```diff
@@ -999,13 +1004,27 @@ impl<T: ?Sized> Shared<T> {
     pub unsafe fn new(ptr: *const T) -> Self {
         Shared { pointer: NonZero::new(ptr), _marker: PhantomData }
     }
-}
 
-#[unstable(feature = "shared", issue = "27730")]
-impl<T: ?Sized> Shared<T> {
+    /// Acquires the underlying `*mut` pointer.
+    pub fn ptr(self) -> *mut T {
+        self.pointer.get() as *mut _
+    }
+
+    /// Dereferences the content.
+    pub unsafe fn get(&self) -> &T {
+        &*self.ptr()
+    }
+
+    /// Mutably dereferences the content.
```

Comment:
Should this be on `Shared` at all? I'd expect interior mutability to be required - or I suppose we could just have something along the lines "you shouldn't ever use this unless there is no other reference around" in the documentation (e.g. `Rc` with a refcount of `1`).